File: ChompMan/UI/HighScoresForm.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Windows.Forms
Imports ChompMan.DataAccess

Namespace UI

    ''' <summary>Displays the top-10 high scores loaded from the Access database.</summary>
    Public Class HighScoresForm
        Inherits Form

        Private WithEvents _btnClose As Button

        ''' <summary>Creates the high-scores screen.</summary>
        Public Sub New()
            InitializeComponent()
            LoadScores()
        End Sub

        Private _listView As ListView

        Private Sub InitializeComponent()
            Me.Text = "ChompMan – High Scores"
            Me.ClientSize = New Size(500, 430)
            Me.FormBorderStyle = FormBorderStyle.FixedDialog
            Me.MaximizeBox = False
            Me.StartPosition = FormStartPosition.CenterParent
            Me.BackColor = Color.Black

            Dim lblTitle As New Label() With {
                .Text = "HIGH SCORES",
                .Font = New Font("Courier New", 20, FontStyle.Bold),
                .ForeColor = Color.Yellow,
                .TextAlign = ContentAlignment.MiddleCenter,
                .AutoSize = False,
                .Size = New Size(500, 50),
                .Location = New Point(0, 10)
            }

            _listView = New ListView() With {
                .View = View.Details,
                .FullRowSelect = True,
                .GridLines = True,
                .BackColor = Color.Black,
                .ForeColor = Color.White,
                .Font = New Font("Courier New", 11),
                .Location = New Point(20, 75),
                .Size = New Size(460, 290)
            }
            _listView.Columns.Add("#", 40)
            _listView.Columns.Add("Name", 130)
            _listView.Columns.Add("Score", 100)
            _listView.Columns.Add("Level", 70)
            _listView.Columns.Add("Date", 110)

            _btnClose = New Button() With {
                .Text = "CLOSE",
                .Font = New Font("Courier New", 12, FontStyle.Bold),
                .ForeColor = Color.Yellow,
                .BackColor = Color.Black,
                .FlatStyle = FlatStyle.Flat,
                .Size = New Size(120, 38),
                .Location = New Point(190, 378)
            }

            Controls.AddRange(New Control() {lblTitle, _listView, _btnClose})
        End Sub

        Private Sub LoadScores()
            Try
                Dim repo As New AccessScoreRepository(Program.DbPath)
                Dim scores = repo.GetTopScores(10)
                PopulateList(scores)
            Catch ex As Exception
                _listView.Items.Add(New ListViewItem(New String() {"", "DB unavailable", ex.Message, "", ""}))
            End Try
        End Sub

        Private Sub PopulateList(scores As List(Of ScoreEntry))
            _listView.Items.Clear()
            For i As Integer = 0 To scores.Count - 1
                Dim s = scores(i)
                Dim item As New ListViewItem((i + 1).ToString())
                item.SubItems.Add(s.PlayerName)
                item.SubItems.Add(s.Score.ToString("N0"))
                item.SubItems.Add(s.LevelReached.ToString())
                item.SubItems.Add(s.PlayedOn.ToLocalTime().ToString("yyyy-MM-dd"))
                If i = 0 Then item.ForeColor = Color.Gold
                _listView.Items.Add(item)
            Next
        End Sub

        Private Sub _btnClose_Click(sender As Object, e As EventArgs) Handles _btnClose.Click
            Close()
        End Sub

    End Class

End Namespace

File: ChompMan/UI/HighScoresForm.cs
using ChompMan.DataAccess;
using System.Drawing;
using System.Windows.Forms;

namespace ChompMan.UI;

/// <summary>Displays the top-10 high scores loaded from the SQLite database.</summary>
public class HighScoresForm : Form
{
    private Button _btnClose;
    private ListView _listView;

    public HighScoresForm()
    {
        InitializeComponent();
        LoadScores();
    }

    private void InitializeComponent()
    {
        Text = "ChompMan – High Scores";
        ClientSize = new Size(500, 430);
        FormBorderStyle = FormBorderStyle.FixedDialog;
        MaximizeBox = false;
        StartPosition = FormStartPosition.CenterParent;
        BackColor = Color.Black;

        var lblTitle = new Label
        {
            Text = "HIGH SCORES",
            Font = new Font("Courier New", 20, FontStyle.Bold),
            ForeColor = Color.Yellow,
            TextAlign = ContentAlignment.MiddleCenter,
            AutoSize = false,
            Size = new Size(500, 50),
            Location = new Point(0, 10)
        };

        _listView = new ListView
        {
            View = View.Details,
            FullRowSelect = true,
            GridLines = true,
            BackColor = Color.Black,
            ForeColor = Color.White,
            Font = new Font("Courier New", 11),
            Location = new Point(20, 75),
            Size = new Size(460, 290)
        };
        _listView.Columns.Add("#", 40);
        _listView.Columns.Add("Name", 130);
        _listView.Columns.Add("Score", 100);
        _listView.Columns.Add("Level", 70);
        _listView.Columns.Add("Date", 110);

        _btnClose = new Button
        {
            Text = "CLOSE",
            Font = new Font("Courier New", 12, FontStyle.Bold),
            ForeColor = Color.Yellow,
            BackColor = Color.Black,
            FlatStyle = FlatStyle.Flat,
            Size = new Size(120, 38),
            Location = new Point(190, 378)
        };
        _btnClose.Click += (_, _) => Close();

        Controls.AddRange(new Control[] { lblTitle, _listView, _btnClose });
    }

    private void LoadScores()
    {
        try
        {
            var repo = new SqliteScoreRepository(Program.DbPath);
            var scores = repo.GetTopScores(10);
            PopulateList(scores);
        }
        catch (Exception ex)
        {
            _listView.Items.Add(new ListViewItem(new[] { "", "DB unavailable", ex.Message, "", "" }));
        }
    }

    private void PopulateList(List<ScoreEntry> scores)
    {
        _listView.Items.Clear();
        for (var i = 0; i < scores.Count; i++)
        {
            var s = scores[i];
            var item = new ListViewItem((i + 1).ToString());
            item.SubItems.Add(s.PlayerName);
            item.SubItems.Add(s.Score.ToString("N0"));
            item.SubItems.Add(s.LevelReached.ToString());
            item.SubItems.Add(s.PlayedOn.ToLocalTime().ToString("yyyy-MM-dd"));
            if (i == 0)
            {
                item.ForeColor = Color.Gold;
            }
            _listView.Items.Add(item);
        }
    }
}
